server/controllers/uploadImage.js

```diff
@@ -0,0 +1,21 @@
+const asyncHandler = require("express-async-handler");
+const { uploadFile } = require('../utils/s3');
+const fs = require('fs');
+const util = require('util');
+const unlinkFile = util.promisify(fs.unlink);
+
+
+// @route POST /uploadImage
+// @desc Upload image to S3
+// @access Private
+exports.uploadImage = asyncHandler(async (req, res, next) => {
+  try {
+    const file = req.file;
+    const result = await uploadFile(file);
+    await unlinkFile(file.path);
+    res.status(200).json({ imagePath: `/images/${result.Key}`});
```

Comment:
It could be okay to return this to the frontend, but keep in mind that we have a field in our `Profile` model that should store this key for us.

So we need to either make sure we follow up with an "update profile" API call on the frontend or implement that logic here in this controller and respond with the updated `Profile` object.